packages/shared/src/components/squads/SquadTypeCard.tsx

```diff
@@ -0,0 +1,63 @@
+import classNames from 'classnames';
+import React, { ReactElement, ReactNode } from 'react';
+import { IconSize } from '../Icon';
+import {
+  AllowedTags,
+  Button,
+  ButtonProps,
+  ButtonSize,
+} from '../buttons/Button';
+import VIcon from '../icons/V';
+
+export type SquadTypeCardProps<ButtonTag extends AllowedTags> = {
+  className?: string;
+  title: ReactNode;
+  description: string;
+  isSelected: boolean;
+  buttonProps?: { text: string } & Omit<
+    ButtonProps<ButtonTag>,
+    'disabled' | 'size' | 'children'
+  >;
+};
+
+export const SquadTypeCard = <ButtonTag extends AllowedTags>({
+  className,
+  title,
+  description,
+  isSelected,
+  buttonProps,
+}: SquadTypeCardProps<ButtonTag>): ReactElement => {
+  const { text: buttonText = 'Select', ...restButtonProps } = buttonProps || {};
+
+  return (
+    <div
+      className={classNames(
+        'flex flex-col flex-1 p-3 rounded-16 justify-between gap-4 relative',
+        isSelected &&
+          'bg-gradient-to-r from-theme-bg-overlay-onion-opacity to-theme-bg-overlay-cabbage-opacity border-2 -m-1 border-theme-divider-primary',
+        className,
+      )}
+    >
+      <div>
+        <h4 className="mb-2 font-bold typo-headline">{title}</h4>
+        <p className="typo-footnote">{description}</p>
+      </div>
+      <Button
+        className="btn-secondary"
+        buttonSize={ButtonSize.XSmall}
+        // @ts-expect-error Property 'disabled' does not exist on type
```

Comment:
Something is off with our types for `Button` it was complaining that `disabled` does not exists, same for `href` in `Details` component. 🤔

Reply:
We should fix that indeed.

Reply:
Fixed in 49e908cf31d6706a8903c279f3079238255c6d8e